okio/src/commonMain/kotlin/okio/ForwardingFilesystem.kt

```diff
@@ -160,7 +160,7 @@ abstract class ForwardingFilesystem(
   override fun list(dir: Path): List<Path> {
     val dir = onPathParameter(dir, "list", "dir")
     val result = delegate.list(dir)
-    return result.map { onPathResult(it, "list") }
+    return result.map { onPathResult(it, "list") }.sorted()
```

Comment:
Shouldn't be needed here.

Reply:
Consider for example a (bad) mapping function that mapped `foo.jpg` to `foo.jpeg`. That'll impact sort order.

I'll add a test that fails without it.